### ISSUE TITLE
during gitlab requests everything stops working

### ISSUE THREAD

**Issue by haarchri** [mvisonneau/gitlab-ci-pipelines-exporter#97]
we have with our own gitlab instance problems - the process starts and the only thing you can see is the following and then nothing more happend in the log but the container is running 

we using gitlab-ci-pipelines-exporter in version 0.3.0 - any ideas what is going on under the hood ? we using gitlab version GitLab Community Edition 12.9.2

```
time="2020-04-30T12:24:50Z" level=info msg="polling metrics from projects refs" count=18
--
  | time="2020-04-30T12:24:50Z" level=debug msg="listing project pipelines" project-id=36054 project-ref=master
  | time="2020-04-30T12:24:50Z" level=debug msg="listing project pipelines" project-id=28373 project-ref=master
  | time="2020-04-30T12:24:50Z" level=debug msg="throttling GitLab requests" for=126.607654ms
  | time="2020-04-30T12:24:50Z" level=debug msg="throttling GitLab requests" for=326.552226ms
  | time="2020-04-30T12:24:50Z" level=debug msg="throttling GitLab requests" for=343.59615ms
  | time="2020-04-30T12:24:51Z" level=debug msg="throttling GitLab requests" for=318.693765ms
```

here is our config

```yaml
gitlab:
  url: https://git.our.own.domain/
  token: $GIT_TOKEN

maximum_gitlab_api_requests_per_second: 5
maximum_projects_poller_workers: 2

#projects to monitor
projects:
  - name: spm/spm-aaa
  - name: spm/spm-zzz
  - name: spm/spm-yyy
  - name: spm/spm-xxx
```

**Comment by mvisonneau** [comment]
Hi @haarchri,

Indeed, there was a critical bug introduced in [0.3.0](https://github.com/mvisonneau/gitlab-ci-pipelines-exporter/releases/tag/0.3.0) which was causing this error, apologies for that. I published a fix in 27fab172681fb450b441cb69d78a364e57af8b9a and released [0.3.1](https://github.com/mvisonneau/gitlab-ci-pipelines-exporter/releases/tag/0.3.1). Can you please updated and let me know if your issue gets sorted?

**Comment by haarchri** [comment]
yes thanks now it is woring ;)
